src/realm/binary_data.hpp

```diff
@@ -57,6 +57,10 @@ class BinaryData {
     template <class T, class A>
     explicit BinaryData(const std::basic_string<char, T, A>&);
 
+    // BinaryData does not store data, callers must manage their own strings.
+    template <class T, class A>
+    BinaryData(const std::basic_string<char, T, A>&&) = delete;
```

Comment:
I think it is an error to have this const. The idea is to move the content out of the argument.